### ISSUE TITLE
Empty sort key on a secondary index causes a useless error that makes it difficult to pinpoint the cause.

### ISSUE THREAD

**Issue by cmm0004** [cashapp/tempest#103]
The original error:
`1) [Guice/ErrorInCustomProvider]: NoSuchElementException: No value present` 
which, sounds like a user error, must have set it up wrong, but "No value present" sounds like an empty java optional, not a guice problem. 
https://cash.slack.com/archives/C02FPF8FL8M/p1647971686751099
here's the full discussion, but it took multiple people and the better part of the day to track down the actual issue

```
internal class V2RawItemTypeFactory : RawItemType.Factory {

  override fun create(tableName: String, rawItemType: KClass<*>): RawItemType {
    val tableSchema = TableSchemaFactory.create<Any>(rawItemType.java)
    return RawItemType(
      rawItemType as KClass<Any>,
      tableName,
      tableSchema.tableMetadata().primaryPartitionKey(),
      tableSchema.tableMetadata().primarySortKey().orElse(null),
      tableSchema.attributeNames().sorted(),
      tableSchema.tableMetadata().indices()
        .filter { it.name() != TableMetadata.primaryIndexName() }
        .map {
          if (it.sortKey().isEmpty) {
            throw IllegalArgumentException("Secondary Index cannot be created, sortKey must be set.")
          }
          ItemType.SecondaryIndex(
            it.name(),
            it.partitionKey().orElse(null)?.name() ?: tableSchema.tableMetadata().primaryPartitionKey(),
            it.sortKey().get().name() 
          )
        }.associateBy { it.name }
    )
  }
}
```
Here: 6th line from the bottom: `it.sortKey().get().name() ` there is no check on the .get(), and if the sort key isn't set (do you always need a sort key on a secondary index? its an Optional<>?) this throws in an unhelpful way.

a change, from me who has no understanding of the system at large - 
```
if (it.sortKey().isEmpty) {
            throw IllegalArgumentException("Secondary Index cannot be created, sortKey must be set.")
          }
```
would at least give back a more useful message in the stack trace, but perhaps the sort key just shouldn't be an Optional? or there should be a logic branch here to handle if it is empty.

**Comment by ShauniArima** [comment]
Is there any update on this part?

**Comment by ShauniArima** [comment]
I have proposed something. Feel free to help by giving examples on how to test :wink:

**Comment by ShauniArima** [comment]
Fixed by #170 :smile:
